Sources/Sentry/SentrySystemEventsBreadcrumbs.m

```diff
@@ -0,0 +1,162 @@
+    NSMutableDictionary<NSString *, NSNumber *> *batteryData = [NSMutableDictionary new];
+    
+    // W3C spec says level must be null if it is unknown
+    if ((currentState != UIDeviceBatteryStateUnknown) || (currentLevel != -1.0)) {
+        float w3cLevel = (currentLevel * 100);
+        batteryData[@"level"] = @(w3cLevel);
+    } else {
+        [SentryLog logWithMessage:@"batteryLevel is unknown."
+                         andLevel:kSentryLogLevelDebug];
+    }
+    
+    batteryData[@"plugged"] = @(isPlugged);
+    return batteryData;
+}
+
+- (void)initOrientationObserver:(UIDevice*)currentDevice
+{
+    if (currentDevice.isGeneratingDeviceOrientationNotifications == NO) {
+        [currentDevice beginGeneratingDeviceOrientationNotifications];
+    }
+    
+    // https://developer.apple.com/documentation/uikit/uideviceorientationdidchangenotification
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(orientationChanged:) name:UIDeviceOrientationDidChangeNotification object:currentDevice];
+    
+    // test
+    // [[NSNotificationCenter defaultCenter] postNotificationName:UIDeviceOrientationDidChangeNotification object:currentDevice];
+}
+
+- (void)orientationChanged:(NSNotification*)notification
+{
+    UIDevice *currentDevice = notification.object;
+    SentryBreadcrumb *crumb =
+    [[SentryBreadcrumb alloc] initWithLevel:kSentryLevelInfo
+                                   category:@"device.orientation"];
+    
+    UIDeviceOrientation currentOrientation = currentDevice.orientation;
+    
+    // Ignore changes in device orientation if unknown, face up, or face down.
+    if (!UIDeviceOrientationIsValidInterfaceOrientation(currentOrientation)) {
+        [SentryLog logWithMessage:@"currentOrientation is unknown."
+                         andLevel:kSentryLogLevelDebug];
+        return;
+    }
+    
+    if (UIDeviceOrientationIsLandscape(currentOrientation)){
+        crumb.data = @{ @"position": @"landscape"};
+    }
+    else {
+        crumb.data = @{ @"position": @"portrait"};
+    }
+    crumb.type = @"navigation";
+    [SentrySDK addBreadcrumb:crumb];
+}
+
+- (void)initKeyboardVisibilityObserver
+{
+    NSNotificationCenter *defaultCenter = [NSNotificationCenter defaultCenter];
+    // https://developer.apple.com/documentation/uikit/uikeyboarddidshownotification
+    [defaultCenter addObserver:self selector:@selector(systemEventTriggered:) name:UIKeyboardDidShowNotification object:nil];
+    
+    // https://developer.apple.com/documentation/uikit/uikeyboarddidhidenotification
+    [defaultCenter addObserver:self selector:@selector(systemEventTriggered:) name:UIKeyboardDidHideNotification object:nil];
+    
+    // test
+    // [[NSNotificationCenter defaultCenter] postNotificationName:UIKeyboardDidHideNotification object:nil];
+}
+
+- (void)systemEventTriggered:(NSNotification*)notification
+{
+    SentryBreadcrumb *crumb =
+    [[SentryBreadcrumb alloc] initWithLevel:kSentryLevelWarning
```

Comment:
M: Are we sure we want to have `kSentryLevelWarning` warning here?
We also add a breadcrumb in case of `UIKeyboardDidShowNotification` not sure if that's worth a warning. More `info` imho.

Reply:
totally agree, my bad